deploy/examples/operator.yaml

```
@@ -107,6 +107,34 @@ data:
   # Labels to add to the CSI RBD Deployments and DaemonSets Pods.
   # ROOK_CSI_RBD_POD_LABELS: "key1=value1,key2=value2"
 
+  # NOTE: Support for defining NodeAffinity for operators other than "In" and "Exists" requires the user to input a
+  # valid v1.NodeAffinity JSON or YAML string.
+  #
+  # For example, the following are valid JSON v1.NodeAffinity strings:
+  # "{"requiredDuringSchedulingIgnoredDuringExecution":{"nodeSelectorTerms":[{"matchExpressions":[{"key":"myKey","operator":"DoesNotExist"}]}]}}"
+  # "{"requiredDuringSchedulingIgnoredDuringExecution":{"nodeSelectorTerms":[{"matchExpressions":[{"key":"myKey","operator":"NotIn","values":["myValue"]}]}]}}"
+  #
+  # For example, the following are valid YAML v1.NodeAffinity strings:
+  # `
+  # ---
+  # requiredDuringSchedulingIgnoredDuringExecution:
+  #   nodeSelectorTerms:
+  #     -
+  #       matchExpressions:
+  #         -
+  #           key: myKey
+  #           operator: DoesNotExist`
+  # `
+  # ---
+  # requiredDuringSchedulingIgnoredDuringExecution:
+  #   nodeSelectorTerms:
+  #     -
+  #       matchExpressions:
+  #         -
+  #           key: myKey
+  #           operator: NotIn
+  #           values:
+  #             - myValue`
   # (Optional) CephCSI provisioner NodeAffinity(applied to both CephFS and RBD provisioner).
```

How about mentioning the new format after the abbreviated format? Since all other rook examples are with yaml, it seems sufficient to just show a yaml example to keep it simple. Also, some minor suggestions to the formatting:
```suggestion
  # (Optional) CephCSI provisioner NodeAffinity (applied to both CephFS and RBD provisioner).
  # CSI_PROVISIONER_NODE_AFFINITY: "role=storage-node; storage=rook, ceph"
  # NOTE: Support for defining NodeAffinity with operators other than "In" and "Exists" requires the user to input a
  # valid v1.NodeAffinity JSON or YAML string. For example:
  # CSI_PROVISIONER_NODE_AFFINITY: |
  #   requiredDuringSchedulingIgnoredDuringExecution:
  #     nodeSelectorTerms:
  #       - matchExpressions:
  #         - key: myKey
  #            operator: NotIn
  #            values:
  #             - myValue
``````

---

Truncated the example to show only one valid YAML string, moved it at the end, and made some formatting changes.